great_expectations/expectations/core/expect_query_results_to_match_source.py

```diff
@@ -275,5 +294,286 @@ def _validate(
+        missing_rows_cols = cls._get_column_names_from_result(missing_rows)
+        unexpected_rows_cols = cls._get_column_names_from_result(unexpected_rows)
+
+        if len(missing_rows_cols) == 1 and len(unexpected_rows_cols) == 1:
+            return cls._create_observed_values_set(
+                configuration=configuration,
+                result=result,
+                runtime_configuration=runtime_configuration,
+                source_col_name=missing_rows_cols[0],
+                target_col_name=unexpected_rows_cols[0],
+            )
+        else:
+            missing_rows_table = cls._create_observed_values_table(missing_rows)
+            unexpected_rows_table = cls._create_observed_values_table(unexpected_rows)
+            return [
+                cls._create_table_rendered_atomic_content(
+                    missing_rows_table,
+                    no_table_msg="No missing rows",
+                ),
+                cls._create_table_rendered_atomic_content(
+                    unexpected_rows_table,
+                    no_table_msg="No unexpected rows",
+                ),
+            ]
+
+    @classmethod
+    def _create_observed_values_set(
+        cls,
+        source_col_name: str,
+        target_col_name: str,
+        configuration: Optional[ExpectationConfiguration] = None,
+        result: Optional[ExpectationValidationResult] = None,
+        runtime_configuration: Optional[dict] = None,
+    ) -> RenderedAtomicContent:
+        result_details = cls._get_details_from_results(result)
+        source_values = [row[source_col_name] for row in result_details["missing_rows"]]
+        target_values = [row[target_col_name] for row in result_details["unexpected_rows"]]
+        renderer_configuration: RendererConfiguration = RendererConfiguration(
+            configuration=configuration,
+            result=result,
+            runtime_configuration=runtime_configuration,
+        )
+        source_param_name = "expected_value"
+        target_param_name = "observed_value"
+        expected_param_prefix = "exp__"
+        ov_param_prefix = "ov__"
+
+        renderer_configuration.add_param(
+            name=source_param_name,
+            param_type=RendererValueType.ARRAY,
+            value=source_values,
+        )
+        renderer_configuration = cls._add_array_params(
+            array_param_name=source_param_name,
+            param_prefix=expected_param_prefix,
+            renderer_configuration=renderer_configuration,
+        )
+
+        renderer_configuration.add_param(
+            name=target_param_name,
+            param_type=RendererValueType.ARRAY,
+            value=target_values,
+        )
+        renderer_configuration = cls._add_array_params(
+            array_param_name=target_param_name,
+            param_prefix=ov_param_prefix,
+            renderer_configuration=renderer_configuration,
+        )
+        observed_value_set = set(target_values)
```

Comment:
I think it's a problem that this will remove dups. I'm not sure if we have an array renderer that handles an unordered array, but we need to be able to render the case where 1 of 2 dups is expected and the other is unexpected because there was only 1 value in the source.